### PR TITLE
Bug 1190578 - Should prompt clear error when generate an application list code in a non-source code repository.

### PR DESCRIPTION
**Pull request by csrwng**
Add a better error message when a git repository cannot be detected.

### CODE REVIEW

File: pkg/generate/generator/sourceref.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (g *SourceRefGenerator) FromDirectory(directory string) (*app.SourceRef, er
 	// Make sure that this is a git directory
 	gitRoot, err := g.repository.GetRootDir(directory)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("could not detect git repository in %s", directory)
```
Comment (smarterclayton) [inline pkg/generate/generator/sourceref.go]: Is the only reason GetRootDir can fail due to the directory not being there?  What if it's permission denied, or a transient filesystem error?

Comment (csrwng) [inline pkg/generate/generator/sourceref.go]: You're right, need better wording

**Comment by csrwng** [comment]
Updated the message

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by smarterclayton** [comment]
transient [merge]

**Pull request merged**